packages/eventual-send/src/local.js

```diff
@@ -0,0 +1,39 @@
+const q = JSON.stringify;
+
+const ntypeof = specimen => (specimen === null ? 'null' : typeof specimen);
+
+export const localApplyFunction = (t, args) => {
+  if (!(t instanceof Function)) {
+    const ftype = ntypeof(t);
+    throw TypeError(
+      `Cannot invoke target as a function; typeof target is ${q(ftype)}`,
+    );
+  }
+  return t(...args);
+};
+
+export const localApplyMethod = (t, method, args) => {
+  if (method === undefined || method === null) {
+    return localApplyFunction(t, args);
+  }
+  if (t === undefined || t === null) {
+    const ftype = ntypeof(t);
+    throw TypeError(
+      `Cannot deliver ${q(method)} to target; typeof target is ${q(ftype)}`,
+    );
+  }
+  const fn = t[method];
+  if (!(fn instanceof Function)) {
+    const ftype = ntypeof(fn);
+    if (ftype === 'undefined') {
```

Comment:
For the diagnostic to be accurate when the property exists with value `undefined`
```suggestion
    if (ftype === 'undefined' && !(method in t)) {
```

Reply:
That fails some values of t, such as `t = true`, with `Uncaught TypeError: Cannot use 'in' operator to search for 'foobar' in true`.

I'm open to other suggestions, though, if you have any for making the diagnostics more accurate.